scanner/indexer/indexer.go

```diff
@@ -19,16 +19,46 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
 	"github.com/quay/claircore"
+	"github.com/quay/claircore/alpine"
 	"github.com/quay/claircore/datastore/postgres"
+	"github.com/quay/claircore/dpkg"
+	"github.com/quay/claircore/gobin"
 	ccindexer "github.com/quay/claircore/indexer"
+	"github.com/quay/claircore/java"
 	"github.com/quay/claircore/libindex"
+	"github.com/quay/claircore/nodejs"
 	"github.com/quay/claircore/pkg/ctxlock"
+	"github.com/quay/claircore/python"
+	"github.com/quay/claircore/rhel"
+	"github.com/quay/claircore/rhel/rhcc"
+	"github.com/quay/claircore/rpm"
+	"github.com/quay/claircore/ruby"
 	"github.com/quay/zlog"
+	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/utils"
 	"github.com/stackrox/rox/scanner/config"
 	"github.com/stackrox/rox/scanner/internal/version"
 )
 
+// ecosystems specifies the package ecosystems to use for indexing.
+func ecosystems(ctx context.Context) []*ccindexer.Ecosystem {
+	es := []*ccindexer.Ecosystem{
+		alpine.NewEcosystem(ctx),
+		dpkg.NewEcosystem(ctx),
+		gobin.NewEcosystem(ctx),
+		java.NewEcosystem(ctx),
+		python.NewEcosystem(ctx),
+		rhcc.NewEcosystem(ctx),
+		rhel.NewEcosystem(ctx),
+		rpm.NewEcosystem(ctx),
+		ruby.NewEcosystem(ctx),
+	}
+	if env.ScannerV4NodeJSSupport.BooleanSetting() {
+		es = append(es, nodejs.NewEcosystem(ctx))
+	}
```

Comment:
Any concerns about this becoming out of sync with the `claircore` defaults or `clair`? ie: if claircore adds support for a new ecosystem and adds it to the defaults, ACS may not have the same support unless we ensure we manually update this.

Reply:
Yes, and we will have to be sure to keep up-to-date with all changes made to ClairCore prior to each release. This is required, however, because we agreed to not set the NodeJS functionality by default in ClairCore

Reply:
Perhaps it's worth modifying claircore so that have the option to add to the default ecosystems instead of full replace so that do not have to watch this so closely.